src/Features/Core/Portable/Diagnostics/EngineV2/InProcCodeAnalysisDiagnosticAnalyzerExecutor.cs

```diff
@@ -13,6 +13,8 @@ namespace Microsoft.CodeAnalysis.Diagnostics.EngineV2
     [ExportWorkspaceService(typeof(ICodeAnalysisDiagnosticAnalyzerExecutor)), Shared]
     internal class InProcCodeAnalysisDiagnosticAnalyzerExecutor : ICodeAnalysisDiagnosticAnalyzerExecutor
     {
+        public static readonly InProcCodeAnalysisDiagnosticAnalyzerExecutor Instance = new InProcCodeAnalysisDiagnosticAnalyzerExecutor();
```

Comment:
If this is going to be a singleton, change the export to be a Factory, and make the actual service class have a private constructor.